metapackages/auto-instrumentations-node/src/utils.ts

```diff
@@ -235,7 +235,11 @@ function getDisabledInstrumentationsFromEnv() {
 export function getResourceDetectorsFromEnv(): Array<Detector | DetectorSync> {
   const resourceDetectors = new Map<
     string,
-    Detector | DetectorSync | Detector[] | DetectorSync[]
+    | Detector
```

Comment:
note for reviewer: this is necessary to allow this single refactor instead of doing all AWS detector at once.

Reply:
Why not do all the AWS detectors at once? What is the gain in opening so many PRs with the exact same refactor?

I personally would prefer to have all detectors in the repo updated in one PR, but at least the ones from the same npm package can be grouped together, no?

Reply:
>Why not do all the AWS detectors at once? What is the gain in opening so many PRs with the exact same refactor?
>
>I personally would prefer...

as you said is more a preference although I find some benefits of splitting. We can discuss each detector independently and allow other detectors to be merged in case one discussion takes long time.

it's my preference but I'm fine in adding up all the changes in a single big PR. I preferred that way to let the owners